src/FCharacter.cpp

```diff
@@ -22,6 +22,12 @@ FCharacter::FCharacter(FVector position, FVector velocity, std::vector<FWeapon>
 	m_type = type;
 }
 
+bool FCharacter::isAlive()
+{
+	//TODO
+	return true;
```

Comment:
I think just in terms of health and different lives, they should be defined with full life value and the life it has left. They can be defined with any value like 100 (or 80, because 20 is lost due to damage) for a normal player and 500 for a boss. When you want to render the health bar, you'll just normalise the life it has left over the full life, i.e 80 / 100 so that you'll know that 1 means full life, where this value will be the texture's one size, and the ratio will show the right health bar.

My guess is they will be defined by using FCharacterType. Once it is defined by this type, you can get information from a group of constant values for all health, projectile, weapon and other damage/health related things (in my opinion, it's better to keep the constant values in the same place). For example, a normal player can do 20 damage, whereas the boss can do 50 damage (of course, this can be varied with weapons and projectiles) and this provides flexibility for different characters/weapons/etc.

What do you think? (btw the boss is you here :D )